Tasks/ServiceFabricDeployV1/Update-DockerSettings.psm1

```diff
@@ -32,9 +32,22 @@ function Update-DockerSettings
             {
                 $dockerRegistryInput = Get-VstsInput -Name dockerRegistryEndpoint -Require
                 $dockerRegistryEndpoint = Get-VstsEndpoint -Name $dockerRegistryInput -Require
-                $userName = $dockerRegistryEndpoint.Auth.Parameters.UserName
-                $password = $dockerRegistryEndpoint.Auth.Parameters.Password
-                $isPasswordEncrypted = $false
+
+                if ($dockerRegistryEndpoint.Data.registrytype -eq "ACR")
+                {
+                    if ($dockerRegistryEndpoint.Auth.Scheme -ne "ServicePrincipal")
+                    {
+                        throw (Get-VstsLocString -Key UnsupportedContainerRegistryAuthScheme -ArgumentList $dockerRegistryEndpoint.Auth.Scheme)
+                    }
+                    $userName = $dockerRegistryEndpoint.Auth.Parameters.ServicePrincipalId
```

Comment:
should we instead query the password from ACR. What are the other tasks consuming this endpoint type doing?

Reply:
Do we get any benefits out of querying password from the ACR given that spn gets created automatically for the ACR resource as soon as we create docker ACR endpoint ?
Docker and Kubernetes tasks are using this docker ACR endpoint to fetch AuthToken using this [class](https://github.com/microsoft/azure-pipelines-tasks/blob/1d75fa8f66aa1cf7a9cb62946939f30f087b2969/Tasks/Common/docker-common/registryauthenticationprovider/acrauthenticationtokenprovider.ts).

Reply:
@bishal-pdMSFT any comments/suggestions here ?

Reply:
Since the SPN being created has permission on ACR resource, it is okay to use SPN credential directly rather than querying password.
The [class](https://github.com/microsoft/azure-pipelines-tasks/blob/1d75fa8f66aa1cf7a9cb62946939f30f087b2969/Tasks/Common/docker-common/registryauthenticationprovider/acrauthenticationtokenprovider.ts) mentioned above also uses the SPN key itself as password.
I believe we need Contributor role for SPN to get password. And contributor role has [permission](https://docs.microsoft.com/en-us/azure/container-registry/container-registry-roles) to do push/pull of images.
The same fact is used by "Azure Resource Manager Service Connection" mode of the registry settings used in this task. See line no 18.